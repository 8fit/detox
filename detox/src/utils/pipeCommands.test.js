const pipeCommands = require('./pipeCommands');

const _win32 = pipeCommands._win32();
const _nix = pipeCommands._nix();

describe('pipeUtils', () => {
  describe('escape in quoted string', () => {
    describe('on win32', () => {
      const escape = _win32.escape.inQuotedString;

      it('should escape double quotes with double quotes', () =>
        expect(escape('Says "Hello"')).toMatchSnapshot());
    });

    describe('on *nix', () => {
      const escape = _nix.escape.inQuotedString;

      it('should escape double quotes with slash', () =>
        expect(escape('Says "Hello"')).toMatchSnapshot());
    });
  });

  describe('escape in quoted regexp', () => {
    describe('on win32', () => {
      const escape = _win32.escape.inQuotedRegexp;

      it('should escape only double quotes', () =>
        expect(escape('^Hello\\s ".*"$')).toMatchSnapshot());
    });

    describe('on *nix', () => {
      const escape = _nix.escape.inQuotedRegexp;

      it('should not escape non-special characters', () =>
        expect(escape('bundle_name')).toMatchSnapshot());

      it('should escape double quotes with slash', () =>
        expect(escape('"a')).toMatchSnapshot());

      it('should escape [\\]', () =>
        expect(escape('[kworker\\0:0]')).toMatchSnapshot());

      it('should escape ^*$', () =>
        expect(escape('^ma*tch$')).toMatchSnapshot());

      it('should escape dots', () =>
        expect(escape('com.company.bundle')).toMatchSnapshot());
    });
  });

  describe('search by regexp pipe command', () => {
    const regexp = '^Hello\\s ".*"$';

    describe('on win32', () => {
      const search = _win32.search.regexp;

      it('should use findstr /R /C:"regexp"', () =>
        expect(search(regexp)).toMatchSnapshot());
    });

    describe('on *nix', () => {
      const search = _nix.search.regexp;

      it('should use grep "regexp"', () =>
        expect(search(regexp)).toMatchSnapshot());
    });
  });

  describe('search by fragment pipe command', () => {
    const fragment = '^Hello\\s ".*"$';

    describe('on win32', () => {
      const search = _win32.search.fragment;

      it('should use findstr /C:"fragment"', () =>
        expect(search(fragment)).toMatchSnapshot());
    });

    describe('on *nix', () => {
      const search = _nix.search.fragment;

      it('should use grep -e "fragment"', () =>
        expect(search(fragment)).toMatchSnapshot());
    });
  });
});
