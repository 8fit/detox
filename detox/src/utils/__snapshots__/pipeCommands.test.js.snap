// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeUtils escape in quoted regexp on *nix should escape [\\] 1`] = `"\\\\[kworker\\\\\\\\0:0\\\\]"`;

exports[`pipeUtils escape in quoted regexp on *nix should escape ^*$ 1`] = `"\\\\^ma\\\\*tch\\\\$"`;

exports[`pipeUtils escape in quoted regexp on *nix should escape dots 1`] = `"com\\\\.company\\\\.bundle"`;

exports[`pipeUtils escape in quoted regexp on *nix should escape double quotes with slash 1`] = `"\\\\\\"a"`;

exports[`pipeUtils escape in quoted regexp on *nix should not escape non-special characters 1`] = `"bundle_name"`;

exports[`pipeUtils escape in quoted regexp on win32 should escape only double quotes 1`] = `"^Hello\\\\s \\"\\".*\\"\\"$"`;

exports[`pipeUtils escape in quoted string on *nix should escape double quotes with slash 1`] = `"Says \\\\\\"Hello\\\\\\""`;

exports[`pipeUtils escape in quoted string on win32 should escape double quotes with double quotes 1`] = `"Says \\"\\"Hello\\"\\""`;

exports[`pipeUtils search by fragment pipe command on *nix should use grep -e "fragment" 1`] = `"grep -e \\"^Hello\\\\s \\\\\\".*\\\\\\"$\\""`;

exports[`pipeUtils search by fragment pipe command on win32 should use findstr /C:"fragment" 1`] = `"findstr /C:\\"^Hello\\\\s \\"\\".*\\"\\"$\\""`;

exports[`pipeUtils search by regexp pipe command on *nix should use grep "regexp" 1`] = `"grep \\"^Hello\\\\s \\\\\\".*\\\\\\"$\\""`;

exports[`pipeUtils search by regexp pipe command on win32 should use findstr /R /C:"regexp" 1`] = `"findstr /R /C:\\"^Hello\\\\s \\"\\".*\\"\\"$\\""`;
